### ISSUE TITLE
Focus/jump to object selected of the currently selected label

### ISSUE THREAD

**Issue by Karol-G** [napari/napari#4108]
Hi,

## 🚀 Feature
Option to focus/jump to the object of the currently selected label. The camera should center on the object and zoom in on it so that the object makes up X% of the view.

## Motivation

Sometimes images have lots of labels that need to be corrected, but it is difficult to keep track on the progress. Being able to automatically jump to the next object in an instance segmentation, correct it and then jump to the next one, would make this task a lot easier.

Best,
Karol

**Comment by Czaki** [comment]
Could You check if the Search Label widget from the PartSeg plugin does this job? 

You could see a video with a presentation on napari hub https://www.napari-hub.org/plugins/PartSeg

(this plugin will not work with dask arrays)

**Comment by Karol-G** [comment]
Hi Czaki,

yes this is what I wanted. However, it would be nice if this could still be implemented in native napari.
I also noticed some issues with the Search Label widget sometimes not working, but I will open a separate issue there on the PartSeg repo.

Best,
Karol

**Comment by Czaki** [comment]
As your description is a little laconic then having a working code is a good starting point. 

Let's wait on one of the core dev team responses. 

The biggest problem with moving my implementation into napari is that it uses C++ code for performance and napari itself is planned to be a pure python. So there needs to be made a decision if create a new package or add such code to one of a  package in the napari ecosystem.

**Comment by tlambert03** [comment]
The feature is certainly nice.  How much code did it take @Czaki?  Can you link to the implementation?

**Comment by Czaki** [comment]
> The feature is certainly nice. How much code did it take @Czaki? Can you link to the implementation?

implementation is here:
https://github.com/4DNucleome/PartSeg/blob/b0f29adee243ba94b1a406c5cf7f118f0aae2cd3/package/PartSeg/plugins/napari_widgets/search_label_widget.py

But it depends on pre-computed information about bounding boxes of Labels (a really good place for throttling/debouncing if we would like to refresh it after edit)

Structure for storage precomputed information is here https://github.com/4DNucleome/PartSeg/blob/b0f29adee243ba94b1a406c5cf7f118f0aae2cd3/package/PartSegCore/roi_info.py#L33 
(important field is `bound_info`)

and Cython backend is here https://github.com/4DNucleome/PartSegCore-compiled-backend/blob/494dee185a3a965f90ba06aacdd1a896d0681fe1/src/PartSegCore_compiled_backend/utils.pyx

The non Cython version is here https://github.com/4DNucleome/PartSeg/blob/b0f29adee243ba94b1a406c5cf7f118f0aae2cd3/package/PartSegCore/roi_info.py#L95-L106 but it is much slower

**Comment by Czaki** [comment]
I would also link #3781 issue which implementation will simplify implementing this feature and allow simple add 3D zooming.

**Comment by XylotrupesGideon** [comment]
I ran into the same question and write this very short function which can be used in the napari console

in my version I had to jump only to the x and y coordinate and then to the first time point where the label appears but it should be very easy to adjust 


```python

import numpy as np

def center(label_number, viewer=viewer, label_layer=viewer.layers["your_label_layer"]):
    """Center the view on the given label number."""
    coordinates = np.argwhere(label_layer.data == label_number)
    if len(coordinates) == 0:
        print(f"Label {label_number} not found.")
        return
    
    # Find the minimum coordinates and the center
    min_coords = np.min(coordinates, axis=0) #for timepoint
    center_coords = np.mean(coordinates, axis=0) #for center coordinates
    
    # Set the camera center (x, y)
    viewer.camera.center = (center_coords[1], center_coords[2])  # (x, y) order

    # Set the first timepoint (z)
    viewer.dims.current_step = (int(min_coords[0]),) + viewer.dims.current_step[1:]
    
    # Set the selected label
    label_layer.selected_label = label_number
    print(f"Centered on label {label_number} at {center_coords}")
```
    
Usage:
    
```python
    center(102)
```

**Comment by psobolewskiPhD** [comment]
Cool! Thanks for sharing!
